airbyte_cdk/sources/declarative/schema/dynamic_schema_loader.py

```diff
@@ -0,0 +1,219 @@
+    ) -> str:
+        """
+        Extracts the key field from the schema using the specified path.
+        """
+        field_key = self._extract_data(raw_schema, field_key_path)
+        if not isinstance(field_key, str):
+            raise ValueError(f"Expected key to be a string. Got {field_key}")
+        return field_key
+
+    def _get_type(
+        self,
+        raw_schema: MutableMapping[str, Any],
+        field_type_path: Optional[List[Union[InterpolatedString, str]]],
+    ) -> Union[Mapping[str, Any], List[Mapping[str, Any]]]:
+        """
+        Determines the JSON Schema type for a field, supporting nullable and combined types.
+        """
+        raw_field_type = (
+            self._extract_data(raw_schema, field_type_path, default="string")
+            if field_type_path
+            else "string"
+        )
+        mapped_field_type = self._replace_type_if_not_valid(raw_field_type)
+        if (
+            isinstance(mapped_field_type, list)
+            and len(mapped_field_type) == 2
+            and all(isinstance(item, str) for item in mapped_field_type)
+        ):
+            first_type = self._get_airbyte_type(mapped_field_type[0])
+            second_type = self._get_airbyte_type(mapped_field_type[1])
+            return {"oneOf": [first_type, second_type]}
+        elif isinstance(mapped_field_type, str):
+            return self._get_airbyte_type(mapped_field_type)
+        else:
+            raise ValueError(
+                f"Invalid data type. Available string or two items list of string. Got {mapped_field_type}."
+            )
+
+    def _replace_type_if_not_valid(
+        self, field_type: Union[List[str], str]
+    ) -> Union[List[str], str]:
+        """
+        Replaces a field type if it matches a type mapping in `types_map`.
+        """
+        if self.schema_type_identifier.types_map:
+            for types_pair in self.schema_type_identifier.types_map:
+                if field_type == types_pair.current_type:
+                    return types_pair.target_type
+        return field_type
+
+    @staticmethod
+    def _get_airbyte_type(field_type: str) -> Mapping[str, Any]:
+        """
+        Maps a field type to its corresponding Airbyte type definition.
+        """
+        if field_type not in AIRBYTE_DATA_TYPES:
+            raise ValueError(f"Invalid Airbyte data type: {field_type}")
+
+        return deepcopy(AIRBYTE_DATA_TYPES[field_type])
+
+    def _extract_data(
+        self,
+        body: Mapping[str, Any],
+        extraction_path: Optional[List[Union[InterpolatedString, str]]] = None,
+        default: Any = None,
+    ) -> Any:
+        """
+        Extracts data from the body based on the provided extraction path.
+        """
+
+        if not extraction_path:
+            return body
+
+        path = [
+            path.eval(self.config) if not isinstance(path, str) else path
+            for path in extraction_path
+        ]
```

Comment:
Can you use different variable names for inner/outer variables to the list comprehension? Just for readability.

e.g. `calculated_path =` or `for current_extraction_path in` or whatever makes sense to you.

Reply:
`path` updated to `node`